packages/core/src/stream/types.ts

```diff
@@ -74,32 +82,37 @@ interface SourcePayload {
   providerMetadata?: SharedV2ProviderMetadata;
 }
 
-interface FilePayload {
+export interface FilePayload {
   data: string | Uint8Array;
   base64?: string;
   mimeType: string;
   providerMetadata?: SharedV2ProviderMetadata;
 }
 
-interface ToolCallPayload {
+export interface ToolCallPayload<TArgs = unknown, TOutput = unknown> {
   toolCallId: string;
   toolName: string;
-  args?: Record<string, any>;
+  input?: TArgs;
```

Comment:
Not 100% sure about this change but I think we should either keep this as args or if we change to input then we should update everywhere that currently uses args to input.